conn.go

```diff
@@ -136,6 +136,293 @@ func NewConnWith(conn net.Conn, config ConnConfig) *Conn {
 	return c
 }
 
+func writeHeader(w *bufio.Writer, clientID string, apiKey apiKey, apiVersion apiVersion, correlationID, size int32) {
```

Comment:
Any reason you couldn't reuse the `(*Conn).writeRequest` method?

Reply:
Nope.  I looked through the write.go and it appeared the headers were being written by hand, but that might have been for performance.  I'll use writeRequest instead.